### ISSUE TITLE
residue 124 in 1sx2 is skipped

### ISSUE THREAD

**Issue by cbaakman** [cmbi/xssp#79]
When generating the dssp file for pdb entry 1sx2, residues 123 and 124 are ignored. This probably has to do with the fact that residue 123 only has one alternate location with identifier B. This should however, not result in residues 123 and 124 missing.

@tonyelewis suggests the following:
```
From what I can see, it looks like the problem is to do with interaction
between AltLoc handling and next-residue detection. In more detail:
  - the atoms in 124 get ignored
  - because at that point, firstAltLoc is set to 'B'
  - because firstAltLoc doesn't get reset at the start of residue 124
  - because the start of residue 124 doesn't get detected
  - because, at that point, atoms.empty()
  - because all atoms in 123 have been skipped
```

**Comment by tonyelewis** [comment]
Thanks for opening this issue. I'll submit a pull request soon.

**Comment by tonyelewis** [comment]
@cbaakman : Please may I ask you to change my contact on the [original description](https://github.com/cmbi/xssp/issues/79#issue-193115137) from my email to @tonyelewis? Thanks very much.

**Comment by tonyelewis** [comment]
Thanks very much.

**Comment by tonyelewis** [comment]
For information... the problem can be replicated using just residues A:120, A:123, A:124 and A:125 and can be simplified down to the backbone atoms:

~~~~~
ATOM   1032  N   MET A 120      23.127   5.241  -5.234  1.00  9.79           N  
ATOM   1033  CA  MET A 120      24.061   4.468  -6.042  1.00  9.61           C  
ATOM   1034  C   MET A 120      24.811   3.451  -5.187  1.00  9.38           C  
ATOM   1035  O   MET A 120      25.054   2.306  -5.620  1.00 10.06           O  
ATOM   1065  N  BGLN A 123      22.567   0.603  -4.409  1.00 11.11           N  
ATOM   1066  CA BGLN A 123      22.431  -0.200  -5.625  1.00 11.15           C  
ATOM   1067  C  BGLN A 123      23.706  -0.950  -5.973  1.00 10.67           C  
ATOM   1068  O  BGLN A 123      23.706  -1.700  -6.955  1.00 11.70           O  
ATOM   1074  N  ALYS A 124      24.772  -0.666  -5.315  0.64  9.24           N  
ATOM   1075  N  BLYS A 124      24.774  -0.724  -5.226  0.36 10.75           N  
ATOM   1076  CA ALYS A 124      26.058  -1.322  -5.576  0.64  9.02           C  
ATOM   1077  CA BLYS A 124      26.116  -1.237  -5.480  0.36 10.92           C  
ATOM   1078  C  ALYS A 124      26.612  -0.964  -6.962  0.64  9.77           C  
ATOM   1079  C  BLYS A 124      26.599  -0.944  -6.904  0.36  9.69           C  
ATOM   1080  O  ALYS A 124      27.304  -1.768  -7.599  0.64 11.47           O  
ATOM   1081  O  BLYS A 124      27.229  -1.776  -7.565  0.36 11.86           O  
ATOM   1092  N   ARG A 125      26.324   0.274  -7.353  1.00  9.58           N  
ATOM   1093  CA  ARG A 125      26.849   0.839  -8.587  1.00 10.21           C  
ATOM   1094  C   ARG A 125      28.105   1.626  -8.216  1.00  9.67           C  
ATOM   1095  O   ARG A 125      28.075   2.852  -8.074  1.00 10.39           O  
~~~~~

**Comment by tonyelewis** [comment]
In [this comment](https://github.com/cmbi/xssp/pull/80#issuecomment-266285060), I describe the results of running the proposed fix in #80 across a 2015 snapshot of the PDB. I include details of the 29 PDBs that the proposed fix affects...

> In some of these cases, the proposed fix restores more residues than in 1sx2 (eg, 4 residues in 3f9s, the whole of chain E in 3adm).

**Comment by tonyelewis** [comment]
Thanks.
